compiler/vm/vm.nim

```diff
@@ -2321,10 +2362,20 @@ proc rawExecute(c: var TCtx, start: int, tos: PStackFrame): TFullReg =
 
     inc pc
 
+proc execute(c: var TCtx, start: int, frame: sink TStackFrame): PNode {.inline.} =
+  assert c.sframes.len == 0
+  c.sframes.add frame
+  result = rawExecute(c, start, 0).regToNode
+  assert c.sframes.len == 1
+  c.sframes.setLen(0)
+
+
 proc execute(c: var TCtx, start: int): PNode =
-  var tos = PStackFrame(prc: nil, comesFrom: 0, next: nil)
+  # XXX: instead of building the object first and then adding it to the list, it could
```

Comment:
It's not a hot loop in my opinion.

For the thought exercise:
Since it's a seq of capacity x, then it's automatically initialized to the zero value per element for each frame. Can we of we set the len to 1 afterwards, then just set the one field to -1, should do it?